src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs

```diff
@@ -953,32 +945,73 @@ protected override void Dispose(bool disposing)
             IJsonWriterAsync asynchronousJsonWriter;
             if (container == null)
             {
-                if (this.concreteJsonWriter != null)
-                {
-                    return this.concreteJsonWriter;
-                }
+                // we don't create a default JsonWriter here because that's already handled by the CreateJsonWriter() method.
+                return null;

```

Comment:
Is `null` the right thing to return? Maybe `return this.jsonWriter`?

Reply:
That check is currently done in the `VerifyJsonWriter()`. I guess I could do that here as well, but I reasoned that it would easier to handle matching the different properties in one place, and then keep these `Create**JsonWriter` as simple `static` helper methods. I realize I forgot to make the make static here.